spec/rails_helper.rb

```diff
@@ -65,6 +65,21 @@ class Analytics
     end
   end
 
+  config.around(:each, type: proc { |t| t != :feature }) do |example|
+    now = Time.zone.now
+    simple_stubs = NamespacedSimpleStubs.new
+    simple_stubs.stub_object(Time, :now) { at(now.to_i) }
+    simple_stubs.stub_object(Date, :today) { jd(now.to_date.jd) }
+    simple_stubs.stub_object(DateTime, :now) do
+      jd(now.to_date.jd, now.hour, now.min, now.sec, Rational(now.utc_offset, 86400))
+    end
+    begin
+      example.run
+    ensure
+      simple_stubs.unstub_all!
+    end
+  end
```

Comment:
are we writing this out because there's no other way to skip `:feature` tests?

Reply:
> are we writing this out because there's no other way to skip `:feature` tests?

Not sure if you're specifically referring to the `type: proc`. Yes, that part is intended to exempt the feature specs from frozen time. The rest of it is the core of the freeze logic though, and essentially replicates what Rails test helpers do internally:

https://github.com/rails/rails/blob/48661542a2607d55f436438fe21001d262e61fec/activesupport/lib/active_support/testing/time_helpers.rb#L148-L150